### ISSUE TITLE
Train JiT-B/32 from scratch: persistent periodic grid and no semantic content

### ISSUE THREAD

**Issue by gcy-hub** [LTH14/JiT#71]
Hi, thank you for releasing JiT. I am training a dense **JiT-B/32** model from scratch on ImageNet-1k, and I would appreciate advice on whether the behavior below is expected at this stage of training or suggests an implementation/configuration problem.

## Problem

The EMA validation samples have no recognizable semantic content. More concerningly, they progressively develop a very regular, repeating mesh/grid pattern with a period that visually matches the 32-pixel patch scale. The grid is weak at 500 steps, visible at 6,000 steps, and much stronger and darker at 9,000 and 12,000 steps.

The images below are representative class-0 samples from four validation points. Each is a 256x256 PNG.

| Global step | Class-0 EMA validation sample |
| --- | --- |
| 500 | <img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/5660e45c-cd0a-4840-ac6e-9b7cd91a1c3e" />|
| 6,000 | <img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/313d62b9-fb37-4cac-96c2-1f36251056f7" />|
| 9,000 | <img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/b4132cb7-d3cc-43ec-90f9-09d9533edc10" />|
| 12,000 | <img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/e032bc69-d1df-49c8-8a10-b54b5429e11b" /> |

I noticed [issue #57](https://github.com/LTH14/JiT/issues/57), which reports visible patch seams for JiT-B/16. Is this possibly the same phenomenon for B/32, or is the early-training behavior shown here expected?

## Training setup

- **Repository/base revision:** cloned from this repository, commit `cbc743a2ada5e9762697da2c83f8c4f8379e8c17`.
- **Model:** `JiT-B/32`, i.e. depth 12, hidden size 768, 12 heads, bottleneck dimension 128, 32 in-context tokens starting at block 4, and patch size 32. The model has 133.19M trainable parameters.
- **Initialization:** training from scratch; no dense or MoE checkpoint was loaded or converted.
- **Hardware:** 4 x NVIDIA A40 (46 GB).
- **Distributed batch:** per-GPU batch size 64, 4 GPUs, gradient accumulation 1, so global batch size 256.
- **Optimization:** bf16 autocast, AdamW (betas `(0.9, 0.95)`), weight decay 0, base/actual LR `5e-5`, 5 warmup epochs, `constant` LR schedule, 200 planned epochs.
- **JiT/flow settings:** `P_mean=-0.8`, `P_std=0.8`, `noise_scale=1.0`, `t_eps=0.05`, label-drop probability 0.1, 1,000 classes.
- **Sampling for the images above:** EMA-1, Heun sampler, 50 sampling steps, CFG 1.0, interval `[0.0, 1.0]`. At each validation point I generate one image for each class 0--99; the table shows class 0 only.
- **Current progress:** the screenshots are at global steps 500, 6,000, 9,000, and 12,000. Training is still continuing beyond 12,000 steps.

## Questions

1. At roughly 12,000 optimizer steps (about 2.5 ImageNet epochs at global batch 256), is it normal for JiT-B/32 samples to have no recognizable semantics and to show a similarly periodic 32-pixel grid?
2. If this is expected early behavior, around what epoch/step range should recognizable structure normally begin to appear for JiT-B/32?
3. If it is not expected, which settings or code paths would you check first? In particular, are there known sensitivity issues involving patch size 32, `noise_scale`, `t_eps`, the time distribution, EMA, or the sampling configuration?
4. Would you recommend continuing this run unchanged, or stopping early to investigate the periodic artifact?
